File: WorkOffice.Contracts/Models/Organization/CompanyStructureModel.cs
﻿using System;
using System.Collections.Generic;
using System.Text;
using System.Text.Json.Serialization;

namespace WorkOffice.Contracts.Models
{
    public class CompanyStructureModel
    {
        public long CompanyStructureId { get; set; }
        public string Name { get; set; }
        public long StructureTypeId { get; set; }
        public string StructureType { get; set; }
        public string Country { get; set; }
        public string Address { get; set; }
        public string ContactPhone { get; set; }
        public string ContactEmail { get; set; }
        public string CompanyHead { get; set; }
        public long? ParentID { get; set; }
        public string Parent { get; set; }
        [JsonIgnore]
        public string Company { get; set; }
        public long ClientId { get; set; }
    }
}

File: WorkOffice.Web/ClientApp/src/app/admin/organization/company-structure/company-structure.model.ts

export interface CompanyStructureModel {
  companyStructureId: number;
  name: string;
  structureTypeId: number;
  structureType: string;
  country: string;
  address: string;
  contactPhone: string;
  contactEmail: string;
  companyHead: string;
  parentID: number;
  parent: string;
}

export interface AddCompanyStructureModel {
  companyStructureId: number;
  name: string;
  structureTypeId: number;
  country: string;
  address: string;
  contactPhone: string;
  contactEmail: string;
  companyHead: string;
  parentID: number;
}
